flipdisc/framework/app.py

```diff
@@ -189,14 +189,21 @@ def setup_input(
         else:
             in_socket.connect(sock_address)
 
-        # TODO(wrigby) Replace ZMQStream with asyncio
-        self._in_stream[cfg] = zmq.eventloop.zmqstream.ZMQStream(in_socket)
-
         # TODO(wrigby): This is quite hacky - we shouldn't be doing this
         callback = getattr(self, "_input_callback_%s" % cfg, None)
         if callback is None:
             raise Exception("Implementation not available for %s configuration" % cfg)
-        self._in_stream[cfg].on_recv(callback, copy=False)
+
+        async def receive_stream(socket, callback):
+            while True:
+                try:
+                    msg = await socket.recv_multipart()
+                    callback(msg)
+                except:
+                    self._log.error('Error waiting for ZMQ stream message')
+                    pass
```

Comment:
Nit: `pass` isn't needed here

```suggestion
                    self._log.error('Error waiting for ZMQ stream message')
```